app/views/pages/i-belong.html.erb

```diff
@@ -1,3 +1,27 @@
+<% if current_user %>
+  <% programme = Programme.i_belong_certificate %>
+  <% state = programme.user_enrolled?(current_user) ? :enrolled : :unenrolled %>
+  <% enrol_path = programme.enrol_path(user_programme_enrolment: { user_id: current_user.id, programme_id: programme.id }) %>
+<% end %>
+
+<% case state %>
+<% when :enrolled %>
+  <% champion_path = '/i-belong-champions-pack' %>
+  <% posters_link_title = 'Request your posters' %>
+  <% posters_path = i_belong_certificate_path %>
+  <% link_method = :get %>
+<% when :unenrolled %>
+  <% champion_path = enrol_path %>
+  <% posters_link_title = 'Enrol to request' %>
+  <% posters_path = enrol_path %>
+  <% link_method = :post %>
+<% else %>
+  <% champion_path = login_path %>
+  <% posters_link_title = 'Log in to request' %>
+  <% posters_path = login_path %>
+  <% link_method = :get %>
+<% end %>
```

Comment:
Firstly, I'm surprised that this doesn't cause an error for being unable to find the `state` method when current_user is nil. I presume the other two variables are not referenced when current_user is nil, but still feels quite fragile.

I'm wondering if for the sake of less mental complexity it would make sense to always define `programme` and the `enrol_path` as the cost of that query is going to be relatively low.


I'm also wondering if we could move these into the controller; I'm not big fan of logic like this sitting in a view, it feels a bit messy.